backend/Models/Groups/group-model.js

```diff
@@ -0,0 +1,434 @@
+      group_id,
+      username,
+    ],
+        function(error, results) {
+          if (error) {
+            reject(error);
+            return;
+          }
+          resolve(results);
+        }
+      );
+    });
+  }
+
+  /**
+  * @param {Object} param0
+  * @param {String} param0.username
+ ` *@param {String} param0.name
+  * @param {String} param0.description
+  * @param {Number} param0.confidential
+  * @param {String} param0.branch
+  * @param {String} param0.semester
+  * @return {Promise}
+  */
+ function customGroup(
+  username,
+  name,
+  description,
+  confidential,
+  branch,
+  semester
+) {
+  return new Promise(async (resolve, reject) => {
+      pool.getConnection(function(error, connection) {
+          if (error) {
+            reject(error);
+          }
+          connection.beginTransaction(function(error) {
+            if (error) {
+              reject(error);
+            }
+            connection.query(
+                `INSERT INTO groups(name,description,confidential,created_by) VALUES (?,?,?,?) WHERE
+                (SELECT COUNT(username) FROM users WHERE (username=? AND admin=1) `,
+              [
+                name,
+                description,
+                confidential,
+                username,
+                username,
+              ],
+              (error, results, fields) => {
+                if (error) {
+                reject(error);
+                  connection.rollback(function(error) {
+                    connection.release();
+                  });
+              return;
+              }
+              let xy = new Promise(function(resolve, reject) {
+                  connection.query(
+                    "INSERT INTO UserGroups(`username`,`group_id`,`admin`)" +
+                      "VALUES(?,?,?)",
+                    [
+                        username,
+                        name,
+                        1,                      
+                    ],
+                    (error, results, fields) => {
+                      if (error) {
+                        reject(error);
+                        return;
+                      }
+                    }
+                  );
+                });
+                try {
+                  await xy;
+                } catch (e) {
+                  connection.rollback(function(error) {
+                    connection.release();
+                    reject(e);
+                  });
+                  error1 = true;
+                }
+               if (error1) {
+                return;
+              }
+            let insertion = new Promise(function(resolve, reject) {
+              connection.query(
+                `INSERT INTO UserGroups(username,group_id)
+                  SELECT username AS u.username , group_id AS g.id , u.branch , u.semester
+                  FROM
+                  users u, groups g
+                  WHERE
+                  u.branch = ? , u.semester = ? AND g.name = ?
```

Comment:
I guess the ```,``` is unintentional? Shouldn't we use AND here?

Reply:
SELECT name, price, photo
FROM drinks, drinks_photos
WHERE drinks.id = drinks_id 
GROUP BY drinks_id

Example

Reply:
Fixed